Mage/Archivist/Archivist.cs

```diff
@@ -0,0 +1,268 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+
+using Newtonsoft.Json.Linq;
+
+public class Archivist : EventEmitter<JObject> {
+	private Mage mage { get { return Mage.instance; } }
+	private Logger logger { get { return mage.logger("archivist"); } }
+
+	// Local cache of all retrieved vault values
+	private Dictionary<string, VaultValue> _cache = new Dictionary<string, VaultValue>();
+
+
+	// Constructor
+	public Archivist () {
+		// Apply changes to cached vault values when applyDiff event is fired
+		mage.eventManager.on ("archivist:applyDiff", (object sender, JToken diff) => {
```

Comment:
yes a diff is, but the event manager emitter doesn't always return objects
if you check it's type is JToken https://github.com/mage/mage-sdk-unity/pull/13/files#diff-ec6ab44a115b6117ac541abb0b4efcdcR3

So we would have to move to a delegate system to be more specific, but then we need to define a delegate for every event we could possibly have, not a bad thing, just more code ticket is already open to look at this and decide.